### PR TITLE
Update Documentation for GCP Cloud SQL Client Authentication

### PR DESCRIPTION
**Pull request by jimbishopp**
Update docs for #9991.

### CODE REVIEW

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 1]
```diff
@@ -66,8 +66,14 @@ Teleport Database service with the following IAM roles:
 - "Service Account Token Creator": will allow Database service to generate
   IAM auth tokens when connecting to the database as the service account user
   we created above.
-- "Cloud SQL Viewer": will allow Database service to automatically download
-  your Cloud SQL instance root CA certificate.
+- Either "Cloud SQL Viewer" or "Cloud SQL Admin":
+  - "Cloud SQL Viewer" will allow Database service to automatically download
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:70]:
```suggestion
  - "Cloud SQL Viewer" will allow the Database Service to automatically download
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 2]
```diff
@@ -66,8 +66,14 @@ Teleport Database service with the following IAM roles:
...
+    your Cloud SQL instance root CA certificate, but does not support client
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:71]:
```suggestion
    your Cloud SQL instance's root CA certificate, but does not support client
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 3]
```diff
@@ -66,8 +66,14 @@ Teleport Database service with the following IAM roles:
...
+    certificate authentication.
+  - "Cloud SQL Admin": will allow Database service to automatically download
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:73]:
```suggestion
  - "Cloud SQL Admin" will allow the Database Service to automatically download
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 4]
```diff
@@ -66,8 +66,14 @@ Teleport Database service with the following IAM roles:
...
+    your Cloud SQL instance root CA certificate and generate an
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:74]:
```suggestion
    your Cloud SQL instance's root CA certificate and generate an
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 5]
```diff
@@ -66,8 +66,14 @@ Teleport Database service with the following IAM roles:
...
+    ephemeral client certificate when the GCP instance is configured to
+    "Allow only SSL connections".
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:76]:
```suggestion
    "Allow only SSL connections."
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 6]
```diff
@@ -79,15 +85,20 @@ Assign it the "Service Account Token Creator" role:
...
+  IAM roles include more permissions than the database agent needs. To further
+  restrict the service account, you can create a role that includes only the
+  following permissions:
   ```ini
   # Used to generate IAM auth tokens when connecting to a database instance.
   iam.serviceAccounts.getAccessToken
   # Used to auto-download the instance's root CA certificate.
   cloudsql.instances.get
+  # Used to generate an ephemeral client certificate when the GCP instance
+  # is configured to "Allow only SSL connections".
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:98]:
```suggestion
  # is configured to "Allow only SSL connections."
```

File: docs/pages/database-access/guides/mysql-cloudsql.mdx [hunk 7]
```diff
@@ -26,9 +26,16 @@ Assign it the "Cloud SQL Admin" role:
...
+  ```ini
+  # Used to download a list of database users.
   cloudsql.users.list
+  # Used to update a user with a one-time password.
   cloudsql.users.update
+  # Used to auto-download the instance's root CA certificate.
+  cloudsql.instances.get
+  # Used to generate an ephemeral client certificate when the GCP instance
+  # is configured to "Allow only SSL connections".
+  cloudsql.sslCerts.createEphemeral
```
Comment (r0mant) [inline docs/pages/database-access/guides/mysql-cloudsql.mdx:38]: Should we say this one is "optional" then?

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 8]
```diff
@@ -79,15 +85,20 @@ Assign it the "Service Account Token Creator" role:
...
+  cloudsql.sslCerts.createEphemeral
```
Comment (r0mant) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:99]: Same as above - mention this one is optional?

**Review by r0mant — COMMENTED** [review]
@jimbishopp @ptgott From the information organization perspective - do you think it'd make sense to also have a separate section in these guides covering specifically "enforce ssl connections" use-case? It would just explain in a couple of sentences that you need an additional GCP permission to generate client certs and the Teleport behavior in this case (that it will check itself whether instance forces ssl connections and attempt to generate client cert).

Reason is, I'm thinking if somebody asks about this use-case, we'll have a link we could give them. Right now it's sort of integrated in the guides which I think is fine but having a dedicated section might also be useful IMO.

File: docs/pages/database-access/guides/postgres-cloudsql.mdx (hunk 1 above)
Comment (jimbishopp) [reply docs/pages/database-access/guides/postgres-cloudsql.mdx:70]: This was copied from the previous release. There is another bullet above that has the same issue. I'll update everything.

File: docs/pages/database-access/guides/mysql-cloudsql.mdx (hunk 7 above)
Comment (jimbishopp) [reply docs/pages/database-access/guides/mysql-cloudsql.mdx]: I had this originally. Don't remember why I removed it. Added it back.

**Comment by jimbishopp** [comment]
> Do you think it'd make sense to also have a separate section in these guides covering specifically "enforce ssl connections" use-case? 

@ptgott @r0mant I went into this thinking I was going to add a new section, but decided to change the existing sections because I couldn't decide where the new section would go. It also felt inconsistent to have another section describing the permissions needed. I also thought about creating a new box specifically for this but realized the appropriate place would be near the existing "SERVICE ACCOUNT PERMISSIONS" box. Adding a new box seemed redundant. I'm open to suggestion.

Keep in mind that if an existing user enables client authentication on the GCP instance without setting the appropriate permissions, they will receive an error messaging when trying to connect informing them of the permission they are missing.

**Comment by ptgott** [comment]
@r0mant @jimbishopp I think the `Admonition` box makes sense, since it provides context for assigning roles to the Database Service service account when we provide those instructions. If we want to make sure customers can view the box, what would you think of modifying the `Admonition` component to generate an ID that we could add to the URL? Currently the component has no `id`.

Just to check, would most users be fine with assigning the "Service Account Token Creator", "Cloud SQL Viewer", and "Cloud SQL Admin" roles without modification? The current use of the `Admonition` box implies that assigning more restrictive permissions to a new role is a special case.

**Review by r0mant — APPROVED** [review]

**Comment by jimbishopp** [comment]
@r0mant @ptgott I may have found a simple solution for having a link to send users. I added `H3` headers to break the "Create Service Account" section into steps. I like how the headers make each step easier to see and now we have a link specifically for "Grant Permissions" we could send users. WDYT?

> Just to check, would most users be fine with assigning the "Service Account Token Creator", "Cloud SQL Viewer", and "Cloud SQL Admin" roles without modification? The current use of the `Admonition` box implies that assigning more restrictive permissions to a new role is a special case.

I would think most users would want to implement the most restrictive permissions required for production. "Cloud SQL Admin" provides more than they need, but makes the guide less complex and easier for a new user to get through quickly when evaluating Teleport. Unfortunately, the new `cloudsql.sslCerts.createEphemeral` permission is only available for built-in GCP roles, of which "Cloud SQL Admin" is the most restrictive. Roman suggested we try to update these guides so they are still accurate if and when GCP allows the ephemeral permission to be assigned to custom roles in the future.

**Review by r0mant — APPROVED** [review]
@jimbishopp I think it's fine.

@ptgott Can you take another look as well so we can merge and get this backported/deployed?

File: docs/pages/database-access/guides/mysql-cloudsql.mdx [hunk 9]
```diff
@@ -5,19 +5,23 @@ description: How to configure Teleport Database Access with GCP Cloud SQL MySQL.
 
 # GCP Cloud SQL MySQL
 
-## Create service account for Teleport Database service
+## Create Teleport Database Service Account
```
Comment (ptgott) [inline docs/pages/database-access/guides/mysql-cloudsql.mdx:8]:
```suggestion
## Create a service account for the Teleport Database Service
```
Let's use sentence case for headings

File: docs/pages/database-access/guides/mysql-cloudsql.mdx [hunk 10]
```diff
@@ -5,19 +5,23 @@ description: How to configure Teleport Database Access with GCP Cloud SQL MySQL.
...
 
 Teleport uses one-time passwords to authenticate with Cloud SQL MySQL. To be
 able to authenticate with a database instance, Teleport must run as a service
 account that has a few of the "Cloud SQL Admin" role permissions. You can create
 a new service account, or modify an existing one to add required permissions.
 
+### Create Service Account
```
Comment (ptgott) [inline docs/pages/database-access/guides/mysql-cloudsql.mdx:15]:
```suggestion
### Create a service account
```

File: docs/pages/database-access/guides/mysql-cloudsql.mdx [hunk 11]
```diff
@@ -5,19 +5,23 @@ description: How to configure Teleport Database Access with GCP Cloud SQL MySQL.
...
+
 If creating a new one, go to the [Service Accounts](https://console.cloud.google.com/iam-admin/serviceaccounts)
 page and create another service account:
 
 ![Create Service Account](../../../img/database-access/guides/cloudsql/service-account-db-service@2x.png)
 
-Assign it the "Cloud SQL Admin" role:
+### Grant Permissions
```
Comment (ptgott) [inline docs/pages/database-access/guides/mysql-cloudsql.mdx:22]:
```suggestion
### Grant permissions
```

File: docs/pages/database-access/guides/mysql-cloudsql.mdx [hunk 12]
```diff
@@ -26,12 +30,21 @@ Assign it the "Cloud SQL Admin" role:
...
   cloudsql.users.update
+  # Used to auto-download the instance's root CA certificate.
+  cloudsql.instances.get
+  # Used to generate an ephemeral client certificate when the GCP instance
+  # is configured to "Allow only SSL connections." (optional)
+  cloudsql.sslCerts.createEphemeral
   ```
 </Admonition>
 
+### Create Key
```
Comment (ptgott) [inline docs/pages/database-access/guides/mysql-cloudsql.mdx:46]:
```suggestion
### Create a key for the service account
```
To use sentence case and be a bit more descriptive

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 13]
```diff
@@ -26,17 +26,21 @@ If it isn't enabled, you can add this flag using the "Edit configuration" dialog
 at the bottom of the Configuration panel. Note, changing this setting may
 require a database instance reboot.
 
-## Create service account for database
+## Create Service Account for the database
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:29]:
```suggestion
## Create a service account for the database
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 14]
```diff
@@ -26,17 +26,21 @@ If it isn't enabled, you can add this flag using the "Edit configuration" dialog
...
 
 Teleport uses service accounts to connect to Cloud SQL databases.
 
+### Create Service Account
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:33]:
```suggestion
### Create a service account
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 15]
```diff
@@ -26,17 +26,21 @@ If it isn't enabled, you can add this flag using the "Edit configuration" dialog
...
+
 Go to the IAM & Admin [Service Accounts](https://console.cloud.google.com/iam-admin/serviceaccounts)
 page and create a new service account:
 
 ![Create Service Account](../../../img/database-access/guides/cloudsql/service-account-db@2x.png)
 
 Press "Create".
 
+### Grant Permissions
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:42]:
```suggestion
### Grant permissions
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 16]
```diff
@@ -45,6 +49,8 @@ authentication:
 
 Press "Done".
 
+### Select Cloud IAM Authentication
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:52]:
```suggestion
### Configure authentication for your service account
```
To make this cohere a bit more with the sequence of steps we've outlined

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 17]
```diff
@@ -58,38 +64,58 @@ Press "Add" and your Users table should look similar to this:
 See [Creating and managing IAM users](https://cloud.google.com/sql/docs/postgres/create-manage-iam-users)
 in Google Cloud documentation for more info.
 
-## Create service account for Teleport Database service
+## Create Teleport Database Service Account
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:67]:
```suggestion
## Create a service account for Teleport Database Service
```
Let's use sentence case here

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 18]
```diff
@@ -58,38 +64,58 @@ Press "Add" and your Users table should look similar to this:
...
-Teleport Database service with the following IAM roles:
+Teleport Database Service.
 
-- "Service Account Token Creator": will allow Database service to generate
-  IAM auth tokens when connecting to the database as the service account user
-  we created above.
-- "Cloud SQL Viewer": will allow Database service to automatically download
-  your Cloud SQL instance root CA certificate.
+
+### Create Service Account
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:73]:
```suggestion
### Create a service account
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 19]
```diff
@@ -58,38 +64,58 @@ Press "Add" and your Users table should look similar to this:
...
 
 If creating a new one, go to the [Service Accounts](https://console.cloud.google.com/iam-admin/serviceaccounts)
 page and create another service account:
 
 ![Create Service Account](../../../img/database-access/guides/cloudsql/service-account-db-service@2x.png)
 
+### Grant Permissions
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:80]:
```suggestion
### Grant permissions
```

File: docs/pages/database-access/guides/postgres-cloudsql.mdx [hunk 20]
```diff
@@ -58,38 +64,58 @@ Press "Add" and your Users table should look similar to this:
...
   # Used to auto-download the instance's root CA certificate.
   cloudsql.instances.get
+  # Used to generate an ephemeral client certificate when the GCP instance
+  # is configured to "Allow only SSL connections." (optional)
+  cloudsql.sslCerts.createEphemeral
   ```
+
 </Admonition>
 
+### Create Key
```
Comment (ptgott) [inline docs/pages/database-access/guides/postgres-cloudsql.mdx:117]:
```suggestion
### Create a key for the service account
```
To use sentence case and be a bit more descriptive

**Review by ptgott — APPROVED** [review]
I've left minor suggestions. Since these are all suggestions, I have approved so you can merge right after you implement them.

**Pull request merged**